ViewLocator causes warning CA1725

### Describe the bug

The template (as well as all code examples I can find) for the default ViewLocator implements `ITemplate.Build` with this signature:
```cs
Control? Build(object? data)
```
but the interface has a different parameter name:
```cs
Control? Build(object? param)
```

This causes warning [CA1725: Parameter names should match base declaration](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1725)

Depending on your code analysis settings, this can cause the build to fail.

### To Reproduce

1. Create a project from the `avalonia.mvvm` template.
2. Add these properties to the project:
  ```xml
  <EnableNETAnalyzers>True</EnableNETAnalyzers>
  <AnalysisLevel>8.0-recommended</AnalysisLevel>
  <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  ```
3. build the project

The build fails with:
```
ViewLocator.cs(11,39): error CA1725: In member Control? ViewLocator.Build(object? data), change parameter name data to param in order to match the identifier as it has been declared in Control? ITemplate<object?, Control?>.
Build(object? param) (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1725)
```


### Avalonia Templates version

11.0.10.1

### Avalonia version

11.1.3

### Additional context

_No response_

--- a/templates/csharp/app-mvvm/ViewLocator.cs
+++ b/templates/csharp/app-mvvm/ViewLocator.cs
@@ -8,19 +8,17 @@ namespace AvaloniaAppTemplate;
 public class ViewLocator : IDataTemplate
 {
 
-    public Control? Build(object? data)
+    public Control? Build(object? param)
     {
-        if (data is null)
+        if (param is null)
             return null;
         
-        var name = data.GetType().FullName!.Replace("ViewModel", "View", StringComparison.Ordinal);
+        var name = param.GetType().FullName!.Replace("ViewModel", "View", StringComparison.Ordinal);
         var type = Type.GetType(name);
 
         if (type != null)
         {
-            var control = (Control)Activator.CreateInstance(type)!;
-            control.DataContext = data;
-            return control;
+            return (Control)Activator.CreateInstance(type)!;
         }
         
         return new TextBlock { Text = "Not Found: " + name };

--- a/templates/csharp/xplat/AvaloniaTest/ViewLocator.cs
+++ b/templates/csharp/xplat/AvaloniaTest/ViewLocator.cs
@@ -7,12 +7,12 @@ namespace AvaloniaTest;
 
 public class ViewLocator : IDataTemplate
 {
-    public Control? Build(object? data)
+    public Control? Build(object? param)
     {
-        if (data is null)
+        if (param is null)
             return null;
 
-        var name = data.GetType().FullName!.Replace("ViewModel", "View", StringComparison.Ordinal);
+        var name = param.GetType().FullName!.Replace("ViewModel", "View", StringComparison.Ordinal);
         var type = Type.GetType(name);
 
         if (type != null)

